### PR TITLE
[NCLSUP-305] Document releaseStorageUrl usage in bacon/pig documentation

### PR DESCRIPTION
**Pull request by jomrazek**
### Checklist:

* [ ] Have you added a note in the [CHANGELOG wiki](https://github.com/project-ncl/bacon/wiki/Changelog) for your change if user-facing?
* [ ] Have you added unit tests for your change?

### CODE REVIEW

File: guide/build-config.md [hunk 1]
```diff
@@ -66,7 +66,7 @@ product:
 ```
 The `version` specifies the product version we are building for, and the `milestone` represents the 'step' at which we are in the version. A typical milestone value is DR1, DR2, etc or ER1, ER2 etc. The combination of the version and milestone forms the full release name of the product: e.g 1.0.0.CR1.
 
-**TODO**: talk about automatic generation of milestone versions with `--releaseStorageUrl`
+The `--releaseStorageUrl` is location of the release storage, typically on rcm-guest staging. Using '*' in your milestone you could also use it for automatic incrementation of milestone value. For example using 'ER*' will check release storage and increment version to next available milestone version.
```
Comment (rnc) [inline guide/build-config.md:70]: I think we should add when it is used (i.e. phase). If '*' isn't used in the milestone, what else is it used for ? (as you said `also` )

Comment (rnc) [reply guide/build-config.md:70]: Sorry , but I still don't find that clear:
1. In `release` I didn't think it does any actual uploading to rcm-guest? i.e. therefore _how_ is it used in `release` stage? - does it generate and alter scripts ?
2. What phase is milestone increment done in? I presume `run`, but its not explicitly said.

In other words, I think its needs to be made explicit how its used in the different phases please.

**Review by rnc — CHANGES_REQUESTED** [review]
See inline comments.

File: guide/build-config.md (hunk 1 above)
Comment (jomrazek) [reply guide/build-config.md:70]: "upload to candidates script" is term from pig release documentation so I used that, I can put it in quotes or use "the"

**Review by rnc — APPROVED** [review]

**Review by thescouser89 — APPROVED** [review]
LGTM

**Pull request merged**